frontend/src/api/gameApi.ts

```diff
@@ -582,3 +588,72 @@ export function markReviewed(idToken: string, cohort: string, id: string) {
         { headers: { Authorization: `Bearer ${idToken}` } },
     );
 }
+
+function isURL(
+    url: string,
+    {
+        hostname,
+        pathParts,
+    }: {
+        hostname: string;
+        pathParts: RegExp[];
+    },
+): boolean {
+    let urlObj: URL | null = null;
+    try {
+        urlObj = new URL(url.trim());
+    } catch (error) {
+        // ...
```

Comment:
This was to reduce redundancy because we'll need to check urlObj anyways to keep Typescript happy. I'll update anyways though, because I can see your point.